db.go

```diff
@@ -349,22 +377,31 @@ type DB struct {
 	IgnoreUnmappedCols bool
 	Context            context.Context
 	Logger             QueryLogger
-	mu                 sync.RWMutex
-	models             map[reflect.Type]*Model
-	mappings           map[reflect.Type]fieldMap
+	// Query deadline, ignored if Deadline.IsZero().
+	Deadline time.Time
+	// Function to add a deadline to a query.
+	DeadlineQueryRewriter func(db *DB, query interface{}, millis int64) (queryWithDeadline interface{}, err error)
```

Comment:
I suspect that it takes some skill to correctly implement this query rewriter. You can do it easily. Could you provide an implementation for some database? How do you expect users of this library to intelligently set this variable?